packages/flutter/lib/src/material/text_field.dart

```diff
@@ -371,6 +381,38 @@ class TextField extends StatefulWidget {
   /// textfield's internal gesture detector, use a [Listener].
   final GestureTapCallback onTap;
 
+  /// Function to generate a custom counter widget. Passed the build context,
```

Comment:
The first "paragraph" should just be one line. Probably best to refer to this as a callback:

Callback that generates a custom [InputDecorator.counter] widget.